sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/LDeviceAdapterTest.java

```diff
@@ -173,7 +176,246 @@ void getLNAdaptersInclundigLN0() {
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        IEDAdapter iedAdapter = sclRootAdapter.getIEDAdapterByName("IED_NAME");
+        iedAdapter.getCurrentElem().getAccessPoint().get(0).setServices(tServices);
+        LDeviceAdapter lDeviceAdapter = iedAdapter.getLDeviceAdapterByLdInst("LD_INS1");
+
+        // When
+        boolean hasCapability = lDeviceAdapter.hasControlBlockCreationCapability(serviceSettingsType);
+        //Then
+        assertThat(hasCapability).isFalse();
+    }
+
+    private static Stream<Arguments> provideHasControlBlockCreationCapabilityFalse() {
+        TServices tServicesLogFix = new TServices();
+        TLogSettings tLogSettingsFix = new TLogSettings();
+        tLogSettingsFix.setCbName(TServiceSettingsNoDynEnum.FIX);
+        tServicesLogFix.setLogSettings(tLogSettingsFix);
+
+        TServices tServicesGseFix = new TServices();
+        TGSESettings tgseSettingsFix = new TGSESettings();
+        tgseSettingsFix.setCbName(TServiceSettingsNoDynEnum.FIX);
+        tServicesGseFix.setGSESettings(tgseSettingsFix);
+
+        TServices tServicesReportFix = new TServices();
+        TReportSettings treportSettingsFix = new TReportSettings();
+        treportSettingsFix.setCbName(TServiceSettingsNoDynEnum.FIX);
+        tServicesReportFix.setReportSettings(treportSettingsFix);
+
+        TServices tServicesSmvFix = new TServices();
+        TSMVSettings tsmvSettingsFix = new TSMVSettings();
+        tsmvSettingsFix.setCbName(TServiceSettingsNoDynEnum.FIX);
+        tServicesSmvFix.setSMVSettings(tsmvSettingsFix);
+
+        return Stream.of(
+            Arguments.of("AccessPoint has no creation capability when LogSetting.cbName=FIX", tServicesLogFix, LOG),
+            Arguments.of("AccessPoint has no creation capability when GseSetting.cbName=FIX", tServicesGseFix, GSE),
+            Arguments.of("AccessPoint has no creation capability when ReportSetting.cbName=FIX", tServicesReportFix, REPORT),
+            Arguments.of("AccessPoint has no creation capability when SmvSetting.cbName=FIX", tServicesSmvFix, SMV)
+        );
+    }
+
+
+    @ParameterizedTest
+    @EnumSource(ServiceSettingsType.class)
+    void hasControlBlockCreationCapability_should_return_false_when_no_existing_services_attribute(ServiceSettingsType serviceSettingsType) throws Exception {
+        // Given
+        SCL scd = SclTestMarshaller.getSCLFromFile("/ied-test-schema-conf/ied_unit_test.xml");
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        IEDAdapter iedAdapter = sclRootAdapter.getIEDAdapterByName("IED_NAME");
+        LDeviceAdapter lDeviceAdapter = iedAdapter.getLDeviceAdapterByLdInst("LD_INS1");
+        iedAdapter.getCurrentElem().getAccessPoint().get(0).setServices(new TServices());
+
+        // When
+        boolean hasCapability = lDeviceAdapter.hasControlBlockCreationCapability(serviceSettingsType);
+        //Then
+        assertThat(hasCapability).isFalse();
+    }
+
```

Comment:
The case in which ServiceSettingsType = null is it tested?

Reply:
No it is missing.
OK, I add a test for hasDataSetCreationCapability and hasControlBlockCreationCapability when ServiceSettingsType == null.